### PR TITLE
ADD ERNIE model

### PR DESCRIPTION
**Pull request by nghuyong**
ERNIE is a serial model released by Baidu including:
ERNIE1.0: [Ernie: Enhanced representation through knowledge integration](https://arxiv.org/abs/1904.09223)
ERNIE2.0: [ERNIE 2.0: A Continual Pre-training Framework for Language Understanding](https://arxiv.org/abs/1907.12412)
ERNIE-tiny: Use data distillation from ERNIE 2.0     
These models have state-of-the-art performances on NLU tasks, especially on Chinese tasks.
So, ERNIE is a very important model in the transformer model family.

Actually, ERNIE model is the same structure as BERT including the above three ERNIE viersions, so we don't add a new model, and just need to convert the weight. 
> Note that: ERNIE2.0 introduces a task embedding, but the official released version doesn't have this embedding weight, and  all released weights are the same with BERT.

I have successfully converted  ERNIE models into PyTorch version and do serial experiments to check the conversion is the same with the original paddlepaddle implement.
More detail: https://github.com/nghuyong/ERNIE-Pytorch

In this PR, I directly add ERNIE model to BERT model related files.
This PR is link to issue: [#issue5117](https://github.com/huggingface/transformers/issues/5117),[#issue928](https://github.com/huggingface/transformers/issues/928) and [#issue514](https://github.com/huggingface/transformers/issues/514)

### CODE REVIEW

**Comment by julien-c** [comment]
We shouldn't have to add those because the library should work out of the box with the models at https://huggingface.co/nghuyong

The discoverability of those models on huggingface.co is a different thing and we're open to suggestions to improve it

For instance, you should upload model cards describing the models including metadata etc. See https://huggingface.co/docs for instance

Pinging @JetRunner and @sshleifer for feedback

**Comment by JetRunner** [comment]
Thanks for your contribution!

Yep, I think this PR is not very necessary here since ERNIE serves as weights for BERT. We have shifted into a "model hub" model so I can't think of a reason to keep hard-coding urls here (and we think we should even remove the existing ones at some point) @julien-c

**Comment by JetRunner** [comment]
Also, what do you mean by "especially on Chinese"? Aren't these checkpoints in English I think?

**Comment by sshleifer** [comment]
We should add some docs and tweet!

**Comment by sshleifer** [comment]
You could copy https://github.com/nghuyong/ERNIE-Pytorch/blob/master/Readme.md to each model card.

**Comment by nghuyong** [comment]
@JetRunner

**Comment by JetRunner** [comment]
> @JetRunner 
> 
> ERNIE1.0 is for Chinese, more performance detail: https://arxiv.org/abs/1904.09223
> 
> ERNIE2.0 and ERNIE-tiny are for English.

Yeah, this raises some confusion here. Thanks for your clarification!

**Comment by nghuyong** [comment]
@julien-c @JetRunner @sshleifer

**Comment by JetRunner** [comment]
Thanks @nghuyong! The model cards look awesome.
We'll take it from here and add some metadata and then merge the model cards.

**Comment by nghuyong** [comment]
@JetRunner  Thanks

**Review by JetRunner — APPROVED** [review]

**Comment by JetRunner** [comment]
Hi @nghuyong ,
I just tried your script, and found that
```
Some weights of BertForMaskedLM were not initialized from the model checkpoint at nghuyong/ernie-1.0 and are newly initialized: ['cls.predictions.transform.LayerNorm.bias', 'cls.predictions.decoder.bias', 'cls.predictions.bias', 'cls.predictions.transform.LayerNorm.weight', 'cls.predictions.transform.dense.bias', 'cls.predictions.decoder.weight', 'cls.predictions.transform.dense.weight']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
```

This error is very common if you upload a model that was loaded by `BertModel`. Could you re-upload these checkpoints by loading them in `BertForMaskedLM` first? That would be something like

```python
tokenizer = BertTokenizer.from_pretrained('./convert')
model = BertForMaskedLM.from_pretrained('./convert')

model.save_pretrained('./saved')
```
This will allow users to directly use the checkpoints to do masking without fine-tuning them. Also, could you help convert the checkpoints to TF as well? We have a tutorial here: https://huggingface.co/transformers/model_sharing.html

**Comment by nghuyong** [comment]
@JetRunner  OK，I will update soon

**Comment by nghuyong** [comment]
@JetRunner have updated now~
